### DISCUSSION TITLE
`The npm ci command can only install with an existing package-lock.json` Error

### DISCUSSION THREAD

**Discussion by seoeaa** [appwrite/appwrite#5124] [Q&A]
I tried to make a stack in Portainer did not take off either

```
 => CACHED [composer 4/5] COPY composer.json /usr/local/src/                                                            0.0s 
 => ERROR [node 4/5] RUN npm ci                                                                                         1.0s 
 => CANCELED [composer 5/5] RUN composer install --ignore-platform-reqs --optimize-autoloader     --no-plugins --no-sc  1.1s 
------
 > [node 4/5] RUN npm ci:
#0 0.927 npm ERR! The `npm ci` command can only install with an existing package-lock.json or
#0 0.928 npm ERR! npm-shrinkwrap.json with lockfileVersion >= 1. Run an install with npm@5 or
#0 0.928 npm ERR! later to generate a package-lock.json file, then try again.
#0 0.929
#0 0.929 npm ERR! A complete log of this run can be found in:
#0 0.930 npm ERR!     /root/.npm/_logs/2023-02-14T19_40_26_793Z-debug-0.log
```

**Comment by stnguyen90** [comment]
@seoeaa, thanks for creating this! 🙏🏼 

It seems you're trying to run Appwrite from source code. Do you really need to run Appwrite from source code? This is typically only needed if you're trying to modify the source code. If you really are, make sure to follow our [Contributing docs](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md#setup-from-source).

If you only need to run Appwrite to build apps, you should follow our [Installation docs](https://appwrite.io/docs/installation). If you want to use Portainer, ensure you're deploying all Appwrite containers and supplying all the necessary environment variables. You can find the necessary docker-compose.yml and .env files [here](https://appwrite.io/docs/installation#manual).
